src/components/visualization-bar/index.js

```diff
@@ -0,0 +1,288 @@
+import { select, div } from 'utils/selection';
+import { mergeDefined, defined } from 'utils/utils';
+import { BarTypes, BarParts, generateGroup } from './lib';
+
+/**
+ * 
+ */
+export class VisualizationBar {
+    constructor(selector, options) {
+        this.percents = {
+            [BarTypes.PROGRESS]: 0,
+            [BarTypes.BUFFER]: 0,
+            [BarTypes.BALANCE]: 100
```

Comment:
From my understanding, this component only supports 3 segments. After speaking to UX, it seems the component should support `n` segments, meaning a lot of the code associated to these `BarTypes` is somewhat redundant